Automattic/wp-calypso
# Jetpack Settings: Add actions, reducer, selectors and query components for module settings

### Introduction

This PR introduces actions, reducer and selectors for managing (retrieving and updating) Jetpack module settings. Also, a query component for managing network requests for module settings is being introduced. This PR is part of the Jetpack Module Settings group effort (#9171).

### New WP.COM API methods
For interacting with the .com API it introduces 2 methods that are not yet implemented on the API side:

* `fetchJetpackModuleSettings()`
* `updateJetpackModuleSettings()`

In order to fully test these methods you can:
* Apply D3588-code on your .com sandbox
* Apply https://github.com/Automattic/jetpack/pull/5418 on your Jetpack site.
* Connect your Jetpack site to your .com sandbox and verify they talk to each other.

### To test

* Checkout this branch
* Play with the `QueryJetpackModuleSettings` query component and verify it dispatches the proper network requests - `<QueryJetpackModuleSettings siteId={ 1234 } moduleSlug="carousel" />`.
* Verify all tests are working: `npm run test-client client/state/jetpack-settings/module-settings/`

/cc @oskosk @roccotripaldi @johnHackworth
